### DISCUSSION TITLE
Re-using programmer hardware

### DISCUSSION THREAD

**Discussion by prandeamus** [felias-fogg/dw-link#3] [General]
Hey! Would it be feasible to dedicate an existing USB ISP programmer as a debugger? I mean something like this: https://www.amazon.co.uk/gp/product/B01L1P5048/ref=ppx_yo_dt_b_asin_title_o08_s00?ie=UTF8&psc=1 but there are clones everywhere. Buying something that is cheap and can have firmware for a dedicated debugger sounds like a great idea.

**Comment by felias-fogg** [comment]
Good question. Would indeed be great. But I fear: no. 
1. They usually come with only an ATmega8, not an ATmega328(P)
2. Because they implement USB in the MCU itself, they have to deal with the USB interrupts that conflict with the ICP interrupts. At least, I was not able to make dw-link work on 32U4 MCUs.

Best,
Bernhard

**Comment by prandeamus** [comment]
Fair enough. I checked my programmer and it was ATMega8A.  Thanks for checking anyway.

**Comment by felias-fogg** [comment]
Hi prandeamus,

I just thought a bit more about it: https://hinterm-ziel.de/index.php/2022/01/13/a-debugwire-hardware-debugger-for-less-than-10-e/

Best,
Bernhard

**Reply by prandeamus** [reply]
I'm glad I was able to get you thinking on the subject. Thanks!
